File: javascript/phone-number/phone-number.js
function isDigit(n) {
  var n = n.toLowerCase();
  return n != ' ' && n >= '0' && n <= '9';
}

var PhoneNumber = function (phoneNumber) {
  this.phoneNumber = phoneNumber;
};

PhoneNumber.prototype.number = function () {
  var numbers = "";
  for (var i = 0; i < this.phoneNumber.length; i++) {
    if (isDigit(this.phoneNumber[i])) {
      numbers += this.phoneNumber[i]
    }
  }
  if (numbers.length == 11 && numbers[0] == '1') {
    numbers = numbers.substr(1, 10);
  }
  return numbers.length != 10 ? "0000000000" : numbers;
};

PhoneNumber.prototype.areaCode = function () {
  return this.number().substr(0, 3);
};

PhoneNumber.prototype.toString = function () {
  return '(' + this.areaCode() + ') ' + this.number().substr(3, 3) + '-' + this.number().substr(6, 4);
};

module.exports = PhoneNumber;

File: csharp/phone-number/PhoneNumber.cs
using System.Linq;

public class PhoneNumber
{
  public PhoneNumber(string aNumber)
  {
    Number = new string(aNumber.Where(c => char.IsDigit(c)).ToArray());
    if (Number.Length != 10)
    {
      Number = Number.Length > 10 && Number[0] == '1' ? Number.Remove(0, 1) : "0000000000";
    }
  }

  public string Number { get; private set; }
  public string AreaCode
  {
    get
    {
      return Number.Substring(0, 3);
    }
  }

  public override string ToString()
  {
    return "(" + AreaCode + ") " + Number.Substring(3, 3) + "-" + Number.Substring(6, 4);
  }
}